pypeit/core/quicklook.py

```diff
@@ -0,0 +1,312 @@
+        for kk, item in enumerate(config_lines):
+            if 'detnum' in item:
+                config_lines.pop(kk)
+
+        # Add in name, slitspatnum
+        ridx = config_lines.index('[rdx]')
+        config_lines.insert(ridx+1, '    slitspatnum = {0}'.format(full_cfg['rdx']['slitspatnum']))
+
+        # this is to avoid that the default detnum (which was introduced for mosaic)
+        # will be passed to the reduction and crash it
+        config_lines.insert(ridx+2, '    detnum = None')
+
+    # Generate PypeIt file
+    config_lines = full_cfg.write()
+    pypeitFile = inputfiles.PypeItFile(config=config_lines, 
+                                       file_paths=calibPypeItFile.file_paths,
+                                       data_table=cut_data,
+                                       setup=calibPypeItFile.setup)
+    pypeitFile.write(science_pypeit)
+
+    # Return
+    return science_pypeit, pypeitFile
+
+
+
+def match_science_to_calibs(science_file:str,
+                            ps_sci:pypeitsetup.PypeItSetup, 
+                            spectrograph,
+                            calib_dir:str):
+    """
+    Match a given science frame to the set of pre-made calibrations
+    in the specified reduction folder. If any exists 
+    
+    Args:
+        science_file (str): Full path to the science file
+        ps_sci (:class:`pypeit.pypeitsetup.PypeItSetup`): 
+        spectrograph (:class:`pypeit.spectrographs.spectrograph.Spectrograph`):
+            Spectrograph objec
+        calib_dir (str): Full path to the calibration directory
+
+    Returns:
+        tuple: str, str
+            Name of PypeIt file for calibrations
+            PypeIt Setup class for the science file
+            Name of setup key
+
+    """
+    # Check file exists
+    if not os.path.isfile(science_file):
+        msgs.error("Your science filename {} does not exist. Check your path".format(science_file))
+
+
+    # Generate the setup dict and yamilfy (yes, this is necessary)
+    setup_dict = {}
+    for key in spectrograph.configuration_keys():
+        setup_dict[key] = ps_sci.fitstbl[key][0]
+    setup_dict = utils.yamlify(setup_dict)
+
+    # Check against existing PypeIt files
+    pypeit_files = glob.glob(os.path.join(
+        calib_dir, f'{spectrograph.name}_*', 
+        f'{spectrograph.name}_calib_*.pypeit'))
+    mtch = []
+    setup_key = None
+    for pypeit_file in pypeit_files:
+        # Read
+        pypeitFile = inputfiles.PypeItFile.from_file(pypeit_file)
+
+        # Check for a match
+        match = True
+        for key in spectrograph.configuration_keys():
+            if setup_dict[key] != pypeitFile.setup[key]:
+                match = False
+        if match:
+            mtch.append(pypeit_file)
+            setup_key = pypeitFile.setup_name
+    # Are we ok?
+    if len(mtch) != 1:
+        embed(header='224 of ql')
+        msgs.error("Matched to zero or more than one setup.  Inconceivable!")
```

Comment:
Can we get a gif to pop up when this is thrown? :)

Reply:
Some way to allow the user to set a parameter somewhere that says "if I encounter this while running, just plow ahead with `mtch[0]` and see what happens" would be useful for real-time, automated use. That is, my gut tells me it is better to fail on a specific science case than on an ambiguity. Happy to be overruled on this one.

Reply:
ok, am only going to barf it if there are zero hits.